### DISCUSSION TITLE
在机器剩余35g内存和20g显存的情况下， 无法完成TensorRT-LLM build

### DISCUSSION THREAD

**Discussion by smartmark-pro** [zai-org/ChatGLM3#768] [Q&A] [answered]
根据教程 https://github.com/THUDM/ChatGLM3/blob/main/tensorrt_llm_demo/README.md
执行 
```
python3 build.py -m chatglm3_6b --output_dir trt_engines/chatglm3_6b/fp16/1-gpu
```
![image](https://github.com/THUDM/ChatGLM3/assets/28584805/62ec88e9-2a02-4c68-9194-9c120e458363)

这是切到显卡上加载发生的，默认在内存上加上只能到第四个部分， 同样发生了killed

注意到已经有人提到类似问题 
https://github.com/NVIDIA/TensorRT-LLM/issues/743
https://github.com/NVIDIA/TensorRT-LLM/issues/102

是否有人知道应该怎么办， 或者机器大概什么配置可以完成这个步骤

**Comment by smartmark-pro** [comment]
目前猜测必须要
![image](https://github.com/THUDM/ChatGLM3/assets/28584805/1a8a095e-ee19-495d-8ca3-b93806e877e5)
这些显卡和较大的内存才有可能成功， 消费级显卡都不行

**Comment by zRzRzRzRzRzRzR** [comment, marked as answer]
你的内存不够啊，内存需要能够执行完整的一个fp32的模型推理你设定的token长度好像，也就是48G？
